synapse/federation/sender/__init__.py

```diff
@@ -422,47 +404,38 @@ async def get_federatable_events_and_destinations(
 
     async def _distribute_events(
         self,
-        per_room_events_and_dests: List[
-            Tuple[str, List[Tuple[EventBase, Collection[str]]]]
-        ],
+        events_and_dests: List[Tuple[EventBase, Collection[str]]],
```

Comment:
I think you can relax this constraint:

```suggestion
        events_and_dests: Iterable[Tuple[EventBase, Collection[str]]],
```